TASK: Move node creation dialog stuff to the Neos.Neos.Ui

mhsdesign opened the pull request:
<!-- 
    Thanks for your contribution, we appreciate it!

    The first section should explain briefly what is changed. 
    Some examples are always nice to showcase the use. 
    The content will be used in the change-logs and addresses 
    developers working with Neos.

    If there are issues regarding the topic of your PR link 
    them here as `related:` or `resolved:`
-->

Will be handled in the ui https://github.com/neos/neos-ui/pull/3519

I thought a lot about where the NodeCreation handlers should reside. After discussion this matter with a few people and also extensively with Wilhelm we are sure they should stay in the Neos.Ui. Thats why we move the handling over there.

All the node creation stuff will be moved to the neos ui, even parts of the node type post processors:

Neos.Neos/Classes/NodeTypePostprocessor/CreationDialogPostprocessor.php
 - -> only contained `showInCreationDialog: true` logic, moved fully to ui
 - -> `\Neos\Neos\Ui\Infrastructure\ContentRepository\CreationDialog\CreationDialogNodeTypePostprocessor`
Neos.Neos/Classes/NodeTypePostprocessor/DefaultPropertyEditorPostprocessor.php
 - -> contained editor mapping for the `creationDialog`, this part was incorporated into the `CreationDialogNodeTypePostprocessor`
 - -> class still exists and is used for the "normal" properties but nothing inside of `ui.creationDialog`

Declarations and configurations in the node types have also been migrated.

**Upgrade instructions**

<!-- 
    Add upgrade instructions for breaking changes. 
    Explain who is affected, what has to be adjusted  
-->

**Review instructions**

<!-- 
    If your change is not explained fully by the first section you can 
    add more details here to help the reviewers understand the change.
    We have to understand what you did, why you did it and how we can 
    verify it works correctly and does no harm.
-->

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

mhsdesign committed:
TASK: Remove declaration of ui nodeCreationHandlers

grebaldi reviewed (approved):
Oh, yes :D :+1:

mhsdesign committed:
TASK: Move `CreationDialogPostprocessor` to Neos.Ui

NodeType configuration, which resides in `ui.creationDialog` is part of the Neos.Neos.Ui

kitsunet reviewed (approved):
Not merging so the UI change comes in at the same time.

mhsdesign committed:
TASK: Move `creationDialog` logic of `DefaultPropertyEditorPostprocessor` to Neos.Ui

There it will reside in the `CreationDialogPostprocessor`

NodeType configuration, which resides in `ui.creationDialog` is part of the Neos.Neos.Ui

mhsdesign committed:
TASK: Document promoted elements term for `showInCreationDialog`

... as the Neos Ui class will be named something with `PromotedElementsCreationHandler`

mhsdesign committed:
TASK: Move last reference of `showInCreationDialog` to Neos.Ui

The translation of the `creationDialog` will also be removed as we will not require it anymore in order to fix this bug:
https://github.com/neos/neos-ui/issues/3509

mhsdesign commented:
Okay i was able to move all node creation stuff to the neos ui, even parts of the node type post processors:

Neos.Neos/Classes/NodeTypePostprocessor/CreationDialogPostprocessor.php
 - -> only contained `showInCreationDialog: true` logic, moved fully to ui
 - -> `\Neos\Neos\Ui\Infrastructure\ContentRepository\CreationDialog\CreationDialogNodeTypePostprocessor`
Neos.Neos/Classes/NodeTypePostprocessor/DefaultPropertyEditorPostprocessor.php
 - -> contained editor mapping for the `creationDialog`, this part was incorporated into the `CreationDialogNodeTypePostprocessor`
 - -> class still exists and is used for the "normal" properties but nothing inside of `ui.creationDialog`
